modules/charon-utils/src/main/java/org/wso2/charon3/utils/usermanager/InMemoryUserManager.java

```diff
@@ -60,18 +57,16 @@ public User createUser(User user, Map<String, Boolean> map)
     }
 
     @Override
-    public User getUser(String id, Map<String, Boolean> map)
-            throws CharonException, BadRequestException, NotFoundException {
-       if (inMemoryUserList.get(id) != null) {
-           return (User) CopyUtil.deepCopy(inMemoryUserList.get(id));
-       } else {
-           throw new NotFoundException("No user with the id : " + id);
-       }
+    public User get(String id, Map<String, Boolean> map) throws NotFoundException {
+        if (inMemoryUserList.get(id) != null) {
+            return (User) CopyUtil.deepCopy(inMemoryUserList.get(id));
+        } else {
+            throw new NotFoundException("No user with the id : " + id);
+        }
     }
 
     @Override
-    public void deleteUser(String id)
-            throws NotFoundException, CharonException, NotImplementedException, BadRequestException {
+    public void delete(String id) throws NotFoundException {
```

Comment:
It is not recommended to change public method signatures as they can be used from some other places as well. We need to preserve the backward compatibility.